cinema/tests/test_movie_api.py

```diff
@@ -157,3 +157,181 @@ def test_image_url_is_shown_on_movie_session_detail(self):
         res = self.client.get(MOVIE_SESSION_URL)
 
         self.assertIn("movie_image", res.data[0].keys())
+
+
+class UnauthenticatedMovieApiTests(TestCase):
+    def setUp(self) -> None:
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class AuthenticatedMovieApiTests(TestCase):
+    def setUp(self) -> None:
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@test.com", password="testpass123"
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_list_movie(self):
+        sample_movie()
+        movie_with_actors_and_genres = sample_movie()
+
+        actor = sample_actor(first_name="Testing", last_name="test")
+        genre = sample_genre(name="test_genre")
+
+        movie_with_actors_and_genres.actors.add(actor)
+        movie_with_actors_and_genres.genres.add(genre)
+
+        result = self.client.get(MOVIE_URL)
+
+        movies = Movie.objects.all()
+
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(result.status_code, status.HTTP_200_OK)
+        self.assertEqual(result.data, serializer.data)
+
+    def test_filter_movies_by_actors(self):
+        movie1 = sample_movie(title="movie1")
+        movie2 = sample_movie(title="movie2")
+        movie3 = sample_movie(title="movie3")
+
+        actor1 = sample_actor(first_name="Testing1", last_name="test")
+        actor2 = sample_actor(first_name="Testing2", last_name="test")
+
+        movie1.actors.add(actor1)
+        movie2.actors.add(actor2)
+
+        result = self.client.get(
+            MOVIE_URL, {"actors": f"{actor1.id}, {actor2.id}"}
+        )
+
+        serializer1 = MovieListSerializer(movie1)
+        serializer2 = MovieListSerializer(movie2)
+        serializer3 = MovieListSerializer(movie3)
+
+        self.assertIn(serializer1.data, result.data)
+        self.assertIn(serializer2.data, result.data)
+        self.assertNotIn(serializer3.data, result.data)
+
+    def test_filter_movies_by_title(self):
+        movie1 = sample_movie(title="movie1")
+        movie2 = sample_movie(title="movie2")
+
+        result = self.client.get(MOVIE_URL, {"title": f"{movie1.title}"})
```

Comment:
doesn't your title have an `str` value? Is it needed here?

Reply:
Evidently it does not...
Fixed xD